### DISCUSSION TITLE
setData as property of useQuery return?

### DISCUSSION THREAD

**Discussion by ftzi** [TanStack/query#3589] [Q&A] [answered]
Hi,

At some actions, I return from the server the new data, like the new user data.

Instead of refetching after the api success and having to wait X ms, I wish there was something like

```tsx
const query = useQuery(...)

const myFunc = useCallback(() => query.setData(await apiCall()), [])
```

Is this already possible?

**Comment by ftzi** [comment, marked as answer]
Found https://react-query.tanstack.com/guides/updates-from-mutation-responses. Not really in the way I wanted, but seems to do the job.

**Reply by TkDodo** [reply]
queryClient.setQueryData is indeed what you want here. It wouldn't really help for mutations of the query would return a setter function because the mutation might live in a completely different component than the query.
